tests/baselines/reference/enumAssignmentCompat3.errors.txt

```diff
@@ -91,54 +95,58 @@ enumAssignmentCompat3.ts(87,1): error TS2322: Type 'First.E' is not assignable t
     abc = secondAbcd; // missing 'd'
     ~~~
 !!! error TS2322: Type 'Abcd.E' is not assignable to type 'First.E'.
-!!! error TS2322:   Property 'd' is missing in type 'First.E'.
+!!! error TS2322:   Type 'E.a' is not assignable to type 'E'.

```

Comment:
This error seems super weird; makes me think the PR isn't quite right.

Reply:
I think this is expected, but not necessarily the most desirable, outcome of the implementation. `Abcd.E` is now treated as the union of its constituents, which is `Abcd.a | Abcd.b | Abcd.c | Abcd.d`. When you try to assign the union to `First.E`, we check that every type in the union is assignable to `First.E`. Since `Abcd.a` is the first type in the union that's not assignable to `First.E`, it's the one we report in the diagnostic.

Reply:
Actually looking at the other test cases I see that this is testing an existing behavior I wasn't aware of. If two enums are defined in different namespaces, have the same name, and have the same members, then the enums are mutually assignable. This is pretty strange to me but I'm sure there must be a reason if there are explicit tests for this  behavior.

Reply:
My first guess is that it means you can use enums defined in different versions of the same library interchangeably.